probe_scraper/scrapers/moz_central_scraper.py

```diff
@@ -62,39 +63,67 @@ def load_tags(channel):
     return data
 
 
-def extract_tag_data(tag_data, channel):
+def extract_major_version(version_str):
+    """
+    Given a version string, e.g. "62.0a1",
+    extract the major version as an int.
+    """
+    search = re.search(r"^(\d+)\.", version_str)
+    if search is not None:
+        return int(search.group(1))
+    else:
+        raise Exception("Invalid version string " + version_str)
+
+
+def extract_tag_version(channel, version_str):
+    """
+    Given a tag, e.g. FIREFOX_65_0_RELEASE,
+    extract the major version as an int.
+    """
+    if channel == "release":
+        return int(version_str.split('_')[1])
+    elif channel in ["beta", "nightly"]:
+        return int(version_str.split('_')[2])
+    else:
+        raise Exception("Unsupported channel " + channel)
+
+
+def adjust_version(channel, version):
+    """
+    We work with tags that are the start of version N.
+    We want to treat those revisions as the end of version N-1 instead.
+    Nightly only has tags of the type FIREFOX_AURORA_NN_BASE, so it doesn't
+    need this.
+    """
+    if channel != "nightly":
+        return version - 1
+    return version
+
+
+def extract_tag_data(tag_data, channel, min_fx_version, max_fx_version):
     tag_regex = CHANNELS[channel]['tag_regex']
     tip_node_id = tag_data["node"]
     tags = filter(lambda t: re.match(tag_regex, t["tag"]), tag_data["tags"])
     results = []
+    latest_version = -1
 
     for tag in tags:
-        version = ""
-        if channel == "release":
-            version = tag["tag"].split('_')[1]
-        elif channel in ["beta", "nightly"]:
-            version = tag["tag"].split('_')[2]
-        else:
-            raise Exception("Unsupported channel " + channel)
-
-        # We work with tags that are the start of version N.
-        # We want to treat those revisions as the end of version N-1 instead.
-        # Nightly only has tags of the type FIREFOX_AURORA_NN_BASE, so it doesn't
-        # need this.
-        if channel != "nightly":
-            version = str(int(version) - 1)
-
-        if int(version) >= MIN_FIREFOX_VERSION:
+        version = extract_tag_version(channel, tag["tag"])
+        version = adjust_version(channel, version)
+        latest_version = max(version, latest_version)
+
+        if (version >= min_fx_version and
+           (max_fx_version is None or version <= max_fx_version)):
             results.append({
                 "node": tag["node"],
                 "version": version,
             })
 
-    results = sorted(results, key=lambda r: int(r["version"]))
+    results = sorted(results, key=lambda r: r["version"])
 
-    # Add tip revision.
-    if tip_node_id != results[-1]["node"]:
-        latest_version = str(int(results[-1]["version"]) + 1)
+    # Add tip revision, if we're including the most recent version
+    if (tip_node_id != results[-1]["node"] and
```

Comment:
Would also like your input on this version logic, now that I've included the `max_fx_version` parameter.

Reply:
The logic looks sound to me. We're basically keeping track of it in this loop instead of referring to it as `results[-1]["version"] + 1`, right?